### PR TITLE
jinja2_native: preserve quotes in strings

### LINKED ISSUE #79083: enabling jinja2 native leads to quotes nested in quotes being stripped
### Summary

Enabling jinja2 native leads to quotes nested in other quotes (single quotes in double quotes or vice versa) being stripped. This stripping is additionally done in a seemingly inconsistent and unpredictable way.

### Issue Type

Bug Report

### Component Name

jinja2

### Ansible Version

```console
$ ansible --version
ansible [core 2.13.3]
  config file = /home/yannik/projects/closed/ansible.cfg
  configured module search path = ['/home/yannik/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/yannik/projects/closed/vendor_collections
  executable location = /usr/local/bin/ansible
  python version = 3.10.7 (main, Sep  7 2022, 00:00:00) [GCC 12.2.1 20220819 (Red Hat 12.2.1-1)]
  jinja version = 3.1.2
  libyaml = True

```

### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
ANSIBLE_NOCOWS(/home/yannik/projects/xxx/ansible.cfg) = True
```

### OS / Environment

fedora 36

### Steps to Reproduce

```yaml (paste below)
- hosts: all
  tasks:
    - copy:
        content: |
          {{ myvar }}
        dest: out1
      vars:
        myvar: '"hello {{ ansible_host }}"'
    - copy:
        content: |
          {{ myvar }}
        dest: out2
      vars:
        myvar: '"hello"'
    - copy:
        content: |
          {{ myvar }}
        dest: out3
      vars:
        myvar: ' "hello"'
    - copy:
        content: |
         a = {{ myvar }}
        dest: out4
      vars:
        myvar: '"hello {{ ansible_host }}"'
    - copy:
        content: |
         a = {{ myvar }}
        dest: out5
      vars:
        myvar: '"hello"'
    - copy:
        content: |
         a = {{ myvar }}
        dest: out6
      vars:
        myvar: ' "hello"'

```

```
ansible-playbook -i localhost, test.yml
cat ~/out*
```

### Expected Results

This is the result with jinja2 native disabled, and which I would generally expect:

```
"hello localhost"
"hello"
 "hello"
a = "hello localhost"
a = "hello"
a =  "hello"
```

### Actual Results

```console
hello localhost
hello
 "hello"
a = hello localhost
a = "hello"
a =  "hello"
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY

Fixes #79083

ci_complete

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

`lib/ansible/template/native_helpers.py`

### CODE REVIEW

**Comment by sivel** [comment]
> mkrizek requested a review from sivel 11 minutes ago

Nope 😉

**Review by sivel — APPROVED** [review]
I've thought through this a few times and feel like this is correct, but it also feels like upstream should get this too.

File: test/integration/targets/jinja2_native_types/test_preserving_quotes.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+        that:
+          - quoted_str == '"hello"'
+          - empty_quoted_str == '""'
+      vars:
+        third_nested_lvl: '"hello"'
+        second_nested_lvl: "{{ third_nested_lvl }}"
+        first_nested_lvl: "{{ second_nested_lvl }}"
+        quoted_str: "{{ first_nested_lvl }}"
+        empty_quoted_str: '""'
+        empty_str: "{{ empty_quoted_str}}"
```
Comment (Yannik) [inline test/integration/targets/jinja2_native_types/test_preserving_quotes.yml:14]: @mkrizek `empty_str` is unused. Did you want to use it for an additional assertion, or in place of `empty_quoted_str == '""'`?

Comment (mkrizek) [reply test/integration/targets/jinja2_native_types/test_preserving_quotes.yml:14]: Good catch, thanks!

**Pull request merged**